crates/core/component/dex/src/component/router/route_and_fill.rs

```diff
@@ -95,16 +88,23 @@ pub trait HandleBatchSwaps: StateWrite + Sized {
             ),
             None => (0u64.into(), delta_2),
         };
+        let epoch = self.get_current_epoch().await.expect("epoch is set");
         let output_data = BatchSwapOutputData {
             height: block_height,
-            epoch_starting_height,
             trading_pair,
             delta_1,
             delta_2,
             lambda_1,
             lambda_2,
             unfilled_1,
             unfilled_2,
+            sct_position_prefix: (
+                u16::try_from(epoch.index).expect("epoch index should be small enough"),
+                u16::try_from(block_height - epoch.start_height)
```

Comment:
Can we add a comment about relative block conversion?

Reply:
Done

Reply:
I'd like to re-pitch adding tct protos: https://github.com/penumbra-zone/penumbra/issues/4118

so other systems like the web code doesn't have to manually re-create this serialization logic happening in the .into()'s